src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Crowdin;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\Dumper\XliffFileDumper;
+use Symfony\Component\Translation\Exception\ProviderException;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Andrii Bodnar <andrii.bodnar@crowdin.com>
+ *
+ * In Crowdin:
+ *  * Filenames refers to Symfony's translation domains;
+ *  * Identifiers refers to Symfony's translation keys;
+ *  * Translations refers to Symfony's translated messages
+ *
+ * @experimental in 5.3
+ */
+final class CrowdinProvider implements ProviderInterface
+{
+    /** @var HttpClientInterface */
+    private $client;
+
+    /** @var LoaderInterface */
+    private $loader;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var XliffFileDumper */
+    private $xliffFileDumper;
+
+    /** @var string */
+    private $defaultLocale;
+
+    /** @var string */
+    private $endpoint;
+
+    /** @var string */
+    private $projectId;
+
+    /** @var array */
+    private $files = [];
```

Comment:
This property seems to be a state kept in the service, and it does not seem to be reset. Can we avoid having a stateful service ?